File: python/find-common-characters.py
from collections import Counter

class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        words = [Counter(word) for word in words]
        res = []
        
        for c in words[0]:
            curr = words[0][c]
            for i in range(1, len(words)):
                curr = min(curr, words[i][c])
                if curr == 0:
                    break
            if curr > 0:
                res.extend([c]*curr)
        return res

File: go/find_common_characters.go
package leetcode

func commonChars(words []string) []string {
	word_dic := make([]map[rune]int, len(words))

	for i, word := range words {
		m := make(map[rune]int)
		for _, c := range word {
			m[c] += 1
		}
		word_dic[i] = m
	}

	var res []string
	for key, val := range word_dic[0] {
		curr := val
		for i := 1; i < len(word_dic); i++ {
			if word_dic[i][key] < curr {
				curr = word_dic[i][key]
				if curr == 0 {
					break
				}
			}
		}
		for i := 0; i < curr; i++ {
			res = append(res, string(key))
		}
	}
	return res

}
